datastore/longhorn.go

```diff
@@ -471,19 +498,18 @@ func (s *DataStore) ValidateV2DataEngine(v2DataEngineEnabled bool) error {
 	return nil
 }
 
-func (s *DataStore) AreAllVolumesDetached() (bool, error) {
+func (s *DataStore) AreAllVolumesDetached(backendStoreDriver longhorn.BackendStoreDriverType) (bool, error) {
```

Comment:
nit: change `longhorn.BackendStoreDriverType` to `longhorn.DataEngineType`.

Reply:
There are CRD and proto fields using BackendStoreDriver. How about still using BackendStoreDriver?

Reply:
oh, then it's not easy for now. Let's improve this later.

Reply:
cc @c3y1huang